hypothesis-python/tests/nocover/test_threading.py

```diff
@@ -210,3 +210,17 @@ def slow_test(n):
     should_sleep = True
     with pytest.raises(DeadlineExceeded):
         slow_test()
+
+
+def test_one_of_branches_lock():
+    # I can't actually get this test to reproduce the race locally.
+    # This should in theory reproduce, but the timings here are very tight.
+    branch_counts = set()
+    s = st.one_of(st.integers(), st.integers(), st.integers())
```

Comment:
define a custom strategy which takes a random length of time to `.validate()`?

Reply:
`.validate` doesnt get called, but I can make `branches` slow. Done